### PR TITLE
Do Openshift address reconciliation in Openshift controller

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Does the Openshift address reconciliation in the Openshift controller. As a sideeffect, this avoids writing a defunct token into Openshift Cluster CRs

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/controller/openshift/resources/controlplane_configmap.go [hunk 1]
```diff
@@ -331,49 +331,49 @@ networkConfig:
...
-  masterURL: "{{ .ClusterURL }}"
-  sessionConfig:
-    sessionMaxAgeSeconds: 300
-    sessionName: ssn
-    sessionSecretsFile: ""
-  templates: null
-  tokenConfig:
-    accessTokenMaxAgeSeconds: 86400
-    authorizeTokenMaxAgeSeconds: 300
+#oauthConfig:
```
Comment (mrIncompetent) [inline api/pkg/controller/openshift/resources/controlplane_configmap.go]: I've fixed the run scripts so they contain the necessary flags which are used here:
https://github.com/kubermatic/kubermatic/pull/3109

Comment (alvaroaleman) [reply api/pkg/controller/openshift/resources/controlplane_configmap.go]: Yeah we should discuss what the real solution looks like. Even after Openshift is done, we shouldn't force ppl to use this if they only care about Kubernetes clusters.

WDYT about altering the template to omit the oauth config when the feature is disabled?

**Comment by alvaroaleman** [comment]
@mrIncompetent Would you be fine with merging this if I remove 9bd2a5b3cbc32839fb2c7fff50e1a5c320cafb7e from the Pull?

**Comment by mrIncompetent** [comment]
> @mrIncompetent Would you be fine with merging this if I remove 9bd2a5b from the Pull?

Yes :)

**Comment by alvaroaleman** [comment]
@mrIncompetent Updated and removed the oauth changes, PTAL

**Pull request merged**